File: CS/Behavioural/Strategy/StrategyPatternRunner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static Pattern.Behavioural.Strategy.StrategyPatternRunner;

namespace Pattern.Behavioural.Strategy
{
	internal static partial class StrategyPatternRunner
    {
        public static void Run()
        {
            Console.WriteLine("Strategy Pattern Runner");
            var inputText = "DE3327050399";
            var charsToExtract = '3';
            var extractTextPositiveStrategy = new ExtractTextPositiveStrategy(inputText, charsToExtract);
            var extractTextNegativeStrategy = new ExtractTextNegativeStrategy(inputText, charsToExtract);

            var strategies = new List<IStrategy>();
            strategies.Add(extractTextPositiveStrategy);
            strategies.Add(extractTextNegativeStrategy);

            foreach (var strategy in strategies)
			{
				strategy.Execute();
                Console.WriteLine(strategy.Name + " = " + strategy.GetOutput());
			}

            Console.WriteLine();
        }
    }
}

File: VB/Behavioural/Strategy/StrategyPatternRunner.vb
Public Class StrategyPatternRunner
	Friend Sub Run()
		Dim input As String
		Dim charToExtract As Char

		input = "DE3327050399"
		charToExtract = "3"c

		Dim extractPositiveStrategy As IStrategy = New ExtractTextPositiveStrategy(input, charToExtract)
		Dim extractNegativeStrategy As IStrategy = New ExtractTextNegativeStrategy(input, charToExtract)

		Dim strategies As List(Of IStrategy) = New List(Of IStrategy)()

		strategies.Add(extractPositiveStrategy)
		strategies.Add(extractNegativeStrategy)

		For Each strategy As IStrategy In strategies
			strategy.Execute()
			Console.WriteLine(strategy.Name + ": " + strategy.GetOutout())
		Next

	End Sub

End Class